test/functional/p2p_invalid_block.py

```diff
@@ -81,7 +82,7 @@ def run_test(self):
 
         node.p2p.send_blocks_and_test([block2], node, success=False, request_block=False, reject_reason='bad-txns-duplicate')
 
-        # Check transactions for duplicate inputs
+        # Check transactions for duplicate inputs (CVE-2018-17144)
```

Comment:
This current test case is not complete to check both aspects of the CVE. Would be great if you added the test case for the money printing part of the CVE.

Reply:
I noticed that, when I was studying this source.
TBH I think the test for the other CVE isn’t complete either, as it only checks that the mutated block is rejected, not that the valid block (with same merkle root) is accepted after that. 
I’ll add fixes for both.

Reply:
Added it in a new commit.